lib/trino-collect/src/test/java/io/trino/collect/cache/TestSafeCaches.java

```diff
@@ -77,12 +79,18 @@ private static void assertThatKeyInvalidationIsImpossible(Cache<Object, Object>
         Object object = new Object();
         assertThatThrownBy(() -> cache.invalidate(object)).isInstanceOf(UnsupportedOperationException.class);
         assertThatThrownBy(() -> cache.invalidateAll(ImmutableList.of())).isInstanceOf(UnsupportedOperationException.class);
+        Map<Object, Object> map = cache.asMap();
+        assertThatThrownBy(() -> map.remove(object)).isInstanceOf(UnsupportedOperationException.class);
+        assertThatThrownBy(() -> map.remove(object, object)).isInstanceOf(UnsupportedOperationException.class);
+        assertThatThrownBy(() -> map.replace(object, object)).isInstanceOf(UnsupportedOperationException.class);
+        assertThatThrownBy(() -> map.replace(object, object, object)).isInstanceOf(UnsupportedOperationException.class);
     }
 
     private static void assertThatClearIsImpossible(Cache<Object, Object> cache)
     {
         assertThatThrownBy(cache::invalidateAll).isInstanceOf(UnsupportedOperationException.class);
-        // TODO test asMap().clear()
+        Map<Object, Object> map = cache.asMap();
+        assertThatThrownBy(map::clear).isInstanceOf(UnsupportedOperationException.class);
```

Comment:
OK to inline the `map`, especially that you use `::` operator, so the `asMap()` invocation isn't part of lambda provided to `assertThatThrownBy`.